Add "Segmentation" tutorial to RTD that mirrors SCT course contents (#2991)
neuropoly/spinalcordtoolbox: documentation/source/user_section/tutorials/spinalcord_segmentation.rst

   :align: right
   :figwidth: 20%

   Centerline detection using OptiC

.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/spinalcord_segmentation/mesh_propagation.png
   :align: right
   :figwidth: 20%

   3D mesh propagation using PropSeg

``sct_propseg`` itself is a single command, but internally it uses three processing steps to segment the spinal cord.

#. Detect the approximate center of the spinal cord automatically using a machine learning-based method (OptiC). This is an initialization step for the core algorithm, PropSeg.
#. Create a coarse 3D mesh by propagating along the spinal cord (PropSeg).
#. Refine the surface of the mesh using small adjustments.

   .. note::

      The centerline detection step is also provided in a standalone script called ``sct_get_centerline``.

Example: T2w image
------------------

.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/spinalcord_segmentation/t2_image.png
  :align: right
  :figwidth: 10%

  t2.nii.gz

From the ``single_subject/data`` folder, navigate to the ``t2`` directory, and verify that it contains a single T2-weighted image:

.. code:: sh

>    $ cd t2

Reviewer: I know it's common to display "$" to show that this is what users need to write, but when doing "quick" copy/paste by double clicking on the line, it includes the "$" and then we need to manually remove it, which is cumbersome. Also I anticipate that some users will include the "$", have an error and won't understand the source of the error.

Suggestion:
- get rid of "$"
- find a way to display "$" "outside" of the copy/pastable field. I've seen some webpage where this was possible. Maybe sphinx has some tools to do that...?

- ``-i``, which indicates the input image (``t2.nii.gz``).
- ``-c``, which indicates the contrast of the image (``t2``).
- ``-qc``, the directory for Quality Control reporting (``~/qc_singleSubj``). QC reports will allow us to evaluate the segmentation slice-by-slice.

During execution, the script will provide status updates as it progress through its various stages.

Inspecting the results using QC
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

When complete, the script will output a command to inspect the results. (**Note:** The exact filepath will vary depending on your filesystem.)

.. code:: sh

   Use the following command to see the results in a browser:
   xdg-open "sct_course_london20/single_subject/data/t2/qc_singleSubj/index.html"

Running this command in your Terminal window will open up a page in your default browser. On this page, the spinal cord is displayed slice by slice. It has also been cropped from the overall anatomical image to provide a quick overview. The segmentation is displayed using a red overlay that can be toggled by repeatedly pressing the right arrow key. More information about QC reporting can be found on the <link to QC reporting> page.

> Inspecting the results using FSLEyes

Reviewer: ```suggestion
Inspecting the results using FSLeyes
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you have fsleyes installed, the script will also output a second command to inspect the results. This will be true for all commands run in this tutorial. (**Note:** The exact filepath will vary depending on your filesystem.)

Reviewer: ```suggestion
If you have `FSLeyes <https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FSLeyes>`_ installed, the script will also output a second command to inspect the results. This will be true for all commands run in this tutorial. (**Note:** The exact filepath will vary depending on your filesystem.)
```

.. code:: sh

   Done! To view results, type:
   fsleyes sct_course_london20/single_subject/data/t2/t2.nii.gz -cm greyscale sct_course_london20/single_subject/data/t2/t2_seg.nii.gz -cm red -a 100.0 &

> As with the Quality Control page, the spinal cord segmentation is displayed in red on top of the anatomical image. Further guidance on the usage of FSLEyes can be found in the `FSL Course <https://fsl.fmrib.ox.ac.uk/fslcourse/lectures/practicals/intro1/index.html>`_.

Reviewer: ```suggestion
As with the Quality Control page, the spinal cord segmentation is displayed in red on top of the anatomical image. Further guidance on the usage of FSLeyes can be found in the `FSL Course <https://fsl.fmrib.ox.ac.uk/fslcourse/lectures/practicals/intro1/index.html>`_.
```

.. figure:: https://raw.githubusercontent.com/spinalcordtoolbox/doc-figures/master/spinalcord_segmentation/t2_fsleyes.png
  :align: center
  :figwidth: 75%

>   The segmented image opened in FSLEyes

Reviewer: ```suggestion
  The segmented image opened in FSLeyes
```

Inspecting the results using QC
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Once again, you may either execute the command given by the script, or simply refresh the QC webpage from the previous examples.

In this case, ``sct_deepseg_sc`` has managed to improve upon the results of ``sct_propseg``.

Choosing between ``sct_propseg`` and ``sct_deepseg_sc``
*******************************************************

Although ``sct_deepseg_sc`` was introduced as a follow-up to the original ``sct_propseg``, choosing between the two is not as straightfoward as it may seem. Neither algorithm is strictly superior in all cases; whether one works better than the other is data-dependent. Given the variation in imaging data (imaging centers, sizes, ages, coil strengths, contrasts, scanner vendors, etc.) SCT recommends to try both algorithms with your pilot scans to evaluate the merit of each on your specific dataset, then stick with a single method throughout your study.

> Development of these approaches is an iterative process, and the data used to develop these approaches evolves over time. If you have input regarding what has worked (or hasn't worked) for you, we would be happy to hear your thoughts in the `Spinal Cord MRI forums <https://forum.spinalcordmri.org/>`_, as it could help to improve the toolbox for future users.

Reviewer: ```suggestion
Development of these approaches is an iterative process, and the data used to develop these approaches evolves over time. If you have input regarding what has worked (or hasn't worked) for you, we would be happy to hear your thoughts in the `SCT forum<http://forum.spinalcordmri.org/c/sct>`_, as it could help to improve the toolbox for future users.
```
